xbmc/filesystem/NFSFile.cpp

```diff
@@ -327,10 +329,22 @@ bool CNfsConnection::Connect(const CURL& url, std::string &relativePath)
     }
     m_exportPath = exportPath;
     m_hostName = url.GetHostName();
-    //read chunksize only works after mount
+
+    // read chunksize only works after mount
     m_readChunkSize = nfs_get_readmax(m_pNfsContext);
     m_writeChunkSize = nfs_get_writemax(m_pNfsContext);
 
+    if (m_readChunkSize == 0)
+    {
+      CLog::Log(LOGDEBUG, "NFS Server did not return max read chunksize - Using 128K default");
+      m_readChunkSize = 128 * 1024; // 128K
```

Comment:
I wonder if it's worth making this an advanced setting.

Reply:
There is already chunksize as advancedsetting, that this works though you need to return 1 here,then FileCache will set it to 64K or to chunksize else. Sucks. Smb returns 128K always.

Reply:
@wsnipex 
It seems like a good idea to me but I will do in a follow up PR to keep this simple for backport.

128K has been chosen (and should be fine for all) because 128K is already used in NFSv3 and also when FileCahe is used with NFSv4.

This code force to use default 128K chunk size when source reports 0:

https://github.com/xbmc/xbmc/blob/56b88b2bf5d82fd2b2feb4df4b35f81e70ca1b89/xbmc/filesystem/FileCache.cpp#L138-L144

Reply:
File chunk size is a bit chaotic because is hardcoded in many places:

https://github.com/xbmc/xbmc/blob/56b88b2bf5d82fd2b2feb4df4b35f81e70ca1b89/xbmc/filesystem/File.cpp#L119

Reply:
> There is already chunksize as advancedsetting

This is chunk size for FileCahe only

We can create a new setting for NFS only (when is not used in combination with FileCache)

Reply:
yes, it's cache only, so won't work when caching is disabled.

adding this in a followup PR is fine for me.